libraries/src/CMS/Component/ComponentHelper.php

```diff
@@ -313,11 +313,14 @@ public static function renderComponent($option, $params = array())
 	{
 		$app = \JFactory::getApplication();
 
-		// Load template language files.
-		$template = $app->getTemplate(true)->template;
-		$lang = \JFactory::getLanguage();
-		$lang->load('tpl_' . $template, JPATH_BASE, null, false, true)
-			|| $lang->load('tpl_' . $template, JPATH_THEMES . "/$template", null, false, true);
+		if (!$app->isClient('api'))
```

Comment:
@mbabker can we move this into the early part of JDocument? Given only JDocument is really template aware